### PR TITLE
Add HParams summary + per-operation profile traces to benchmark runs

### PR DESCRIPTION
**Pull request by mridul-sahu**
## Summary

- Each benchmark run now emits a TensorBoard **HParams summary** recording the run's options, populating TB's Parallel Coordinates view so option-value sweeps render as a draggable axis chart instead of needing manual per-axis comparison.
- Each measured save/load operation is wrapped in `jax.profiler.start_trace` / `stop_trace`, routing trace data into the run's TB directory under the Profile plugin's discovery layout (`<output_dir>/tensorboard/plugins/profile/<run>__<operation>`). The Profile tab can then step through any run with no extra plumbing. Tracing is opt-in via `enable_trace` and first-repeat-only by default (`trace_every_repeat` to override).

## Test plan

- [x] `core/core_trace_path_test.py` + `core/metric_test.py` (23 tests) pass
- [x] Full `benchmarks/` suite passes (289 passed, no sibling regressions)
- [x] HParams + per-operation scalars render in TensorBoard (Table + Parallel Coordinates views)

### CODE REVIEW

File: checkpoint/orbax/checkpoint/_src/testing/benchmarks/core/metric.py [hunk 1]
```diff
@@ -476,6 +471,28 @@ def __exit__(self, *exc):
...
 
+def _options_to_hparams(options: Any) -> dict[str, bool | int | float | str]:
+  """Flattens a benchmark options object into a TB HParams-acceptable dict.
+
+  HParams values must be primitives (bool / int / float / str). Anything else
+  (None, list, tuple, nested) is rendered via str() so the run still appears
+  in the Parallel Coordinates view rather than getting dropped.
+  """
+  if dataclasses.is_dataclass(options):
+    raw = dataclasses.asdict(options)
```
Comment (servusdei2018) [inline checkpoint/orbax/checkpoint/_src/testing/benchmarks/core/metric.py:482]: `dataclasses.asdict(options)` performs a recursive deep copy and dict conversion. Since you're flattening complex values to strings anyway in the loop below, `vars(options)` or `options.__dict__` would be a more performant shallow alternative.

**Review by servusdei2018 — CHANGES_REQUESTED** [review]
